wpcom-plugins.php

```diff
@@ -22,6 +22,35 @@ function wpcom_vip_legacy_load_plugin( $plugin = false, $folder = false, $versio
 	return wpcom_vip_load_plugin( $plugin, $folder );
 }
 
+/**
+ * Fix plugin url paths when relative to the theme's directory.
+ *
+ * In plugins_url(), plugin_basename() enforces a directory relative to WP_PLUGIN_DIR or WPMU_PLUGIN_DIR.
+ * This doesn't work out well in cases where "plugins" are still being used in themes/theme-name/plugins.
+ *
+ * Setting a high priority (100) to allow clients to resolve this problem before our fix belows needs to get involved.
+ *
+ * @param string $url    The complete URL to the plugins directory including scheme and path.
+ * @param string $path   Path relative to the URL to the plugins directory. Blank string if no path is specified.
+ * @param string $plugin The plugin file path to be relative to. Blank string if no plugin is specified.
+ *
+ * @return $url
+ */
+function vip_wpcom_compat_fix_theme_relative_plugins_url( $url, $path, $plugin ) {
+	if ( empty( $plugin ) || ! is_string( $plugin ) || false === strpos( $url, '/var/www/wp-content/themes/' ) ) {
```

Comment:
What's the condition where `$plugin` might not be a string?

Reply:
User error. I just noticed that core checks for it here: https://developer.wordpress.org/reference/functions/plugins_url/

Though it does seem safe to assume we are getting a string, even if empty. Notably there is no check here already: https://github.com/Automattic/vip-go-mu-plugins/blob/d1361162e2436f81bcc7493a16af073dd4cb3914/z-client-mu-plugins.php#L84-L113, so perhaps we can just remove in this fix as well.